src/dmd/backend/cod1.d

```diff
@@ -4206,6 +4214,27 @@ void pushParams(ref CodeBuilder cdb, elem* e, uint stackalign, tym_t tyf)
         }
     }
 
+    version (MARS)
+    {
+        case OPstrctor:
+        {
```

Comment:
What is this block of code doing?

Reply:
The same thing it does for DMC, see line 4172.

Reply:
This does the work for the Win32 ABI. It fills `OPstrthis` nodes with the offset of the stack arguments, we need this to be able to call the copy contractor directly on the area here arguments are pushed. DMC does this with non-POD C++ arguments and it works.